src/Raven.Server/ServerWide/ClusterStateMachine.License.cs

```diff
@@ -0,0 +1,206 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Raven.Client.Exceptions.Commercial;
+using Raven.Client.ServerWide;
+using Raven.Server.Commercial;
+using Raven.Server.Json;
+using Raven.Server.ServerWide.Commands;
+using Raven.Server.ServerWide.Commands.ETL;
+using Raven.Server.ServerWide.Commands.Indexes;
+using Raven.Server.ServerWide.Commands.PeriodicBackup;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.ServerWide;
+
+public sealed partial class ClusterStateMachine
+{
+    private const int MinBuildVersion54118 = 54_118;
+
+    private static readonly List<string> _licenseLimitsCommandsForCreateDatabase = new()
+    {
+        nameof(PutIndexCommand),
+        nameof(PutIndexesCommand),
+        nameof(UpdatePeriodicBackupCommand),
+        nameof(UpdatePullReplicationAsSinkCommand),
+        nameof(UpdatePullReplicationAsHubCommand),
+        nameof(UpdateExternalReplicationCommand),
+        nameof(AddRavenEtlCommand),
+        nameof(AddSqlEtlCommand),
+        nameof(EditTimeSeriesConfigurationCommand),
+        nameof(EditDocumentsCompressionCommand),
+        nameof(AddElasticSearchEtlCommand),
+        nameof(AddOlapEtlCommand),
+        nameof(AddQueueEtlCommand)
+    };
+
+    private void AssertLicenseLimits(string type, ServerStore serverStore, DatabaseRecord databaseRecord, ClusterOperationContext context)
```

Comment:
I think we should check only the requested change and not everything of the given type.

Reply:
maybe it would be easier to add an abstract method of `AssertLicenseLimits` to `UpdateDatabaseCommand`

Reply:
We need to call this from addDatabase, when the command is addDatabaseCommand for all the commands in the list(as strings)  so we won't have a way to call it if it is at UpdateDatabaseCommand

Reply:
`AddDatabaseCommand` is a special case, so we can add `AssertLicenseLimits` to it as well.
We can also create an interface `ILicenseLimits` or something.. @grisha-kotler thoughts?

Reply:
ping

Reply:
Done